File: app_Web/src/interface/entity/product-sum.ts
/**
 * 产品汇总表
 *
 * @export
 * @interface ProductSum
 */
export interface ProductSum {

    /**
     * 主键标识
     *
     * @returns {*}
     * @memberof ProductSum
     */
    id?: any;

    /**
     * 产品名称
     *
     * @returns {*}
     * @memberof ProductSum
     */
    name?: any;

    /**
     * 产品负责人
     *
     * @returns {*}
     * @memberof ProductSum
     */
    po?: any;

    /**
     * 计划
     *
     * @returns {*}
     * @memberof ProductSum
     */
    plan?: any;

    /**
     * 开始日期
     *
     * @returns {*}
     * @memberof ProductSum
     */
    begin?: any;

    /**
     * 结束日期
     *
     * @returns {*}
     * @memberof ProductSum
     */
    end?: any;

    /**
     * 草稿
     *
     * @returns {*}
     * @memberof ProductSum
     */
    waitstorycnt?: any;

    /**
     * 激活
     *
     * @returns {*}
     * @memberof ProductSum
     */
    activestorycnt?: any;

    /**
     * 已变更
     *
     * @returns {*}
     * @memberof ProductSum
     */
    changedstorycnt?: any;

    /**
     * 已关闭
     *
     * @returns {*}
     * @memberof ProductSum
     */
    closedstorycnt?: any;

    /**
     * 总计
     *
     * @returns {*}
     * @memberof ProductSum
     */
    storycnt?: any;

    /**
     * Bug数
     *
     * @returns {*}
     * @memberof ProductSum
     */
    bugcnt?: any;

    /**
     * 未开始阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    waitstagestorycnt?: any;

    /**
     * 已计划阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    planedstagestorycnt?: any;

    /**
     * 已立项阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    projectedstagestorycnt?: any;

    /**
     * 研发中阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    developingstagestorycnt?: any;

    /**
     * 研发完毕阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    developedstagestorycnt?: any;

    /**
     * 测试中阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    testingstagestorycnt?: any;

    /**
     * 测试完毕阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    testedstagestorycnt?: any;

    /**
     * 已验收阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    verifiedstagestorycnt?: any;

    /**
     * 已发布阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    releasedstagestorycnt?: any;

    /**
     * 已关闭阶段需求数量
     *
     * @returns {*}
     * @memberof ProductSum
     */
    closedstagestorycnt?: any;

    /**
     * 未开始阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    waitstagestoryhours?: any;

    /**
     * 已关闭阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    closedstagestoryhours?: any;

    /**
     * 已发布阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    releasedstagestoryhours?: any;

    /**
     * 已验收阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    verifiedstagestoryhours?: any;

    /**
     * 测试完毕阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    testedstagestoryhours?: any;

    /**
     * 测试中阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    testingstagestoryhours?: any;

    /**
     * 研发完毕阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    developedstagestoryhours?: any;

    /**
     * 研发中阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    developingstagestoryhours?: any;

    /**
     * 已立项阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    projectedstagestoryhours?: any;

    /**
     * 已计划阶段需求工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    planedstagestoryhours?: any;

    /**
     * 总工时
     *
     * @returns {*}
     * @memberof ProductSum
     */
    totalhours?: any;

    /**
     * 代码错误
     *
     * @returns {*}
     * @memberof ProductSum
     */
    codeerror?: any;

    /**
     * 配置相关
     *
     * @returns {*}
     * @memberof ProductSum
     */
    config?: any;

    /**
     * 安装部署
     *
     * @returns {*}
     * @memberof ProductSum
     */
    install?: any;

    /**
     * 安全相关
     *
     * @returns {*}
     * @memberof ProductSum
     */
    security?: any;

    /**
     * 性能问题
     *
     * @returns {*}
     * @memberof ProductSum
     */
    performance?: any;

    /**
     * 标准规范
     *
     * @returns {*}
     * @memberof ProductSum
     */
    standard?: any;

    /**
     * 测试脚本
     *
     * @returns {*}
     * @memberof ProductSum
     */
    automation?: any;

    /**
     * 设计缺陷
     *
     * @returns {*}
     * @memberof ProductSum
     */
    designdefect?: any;

    /**
     * 其他
     *
     * @returns {*}
     * @memberof ProductSum
     */
    others?: any;

    /**
     * bug总计
     *
     * @returns {*}
     * @memberof ProductSum
     */
    bugsum?: any;
}

File: ibizpms-core/src/main/java/cn/ibizlab/pms/core/ibiz/domain/ProductSum.java
package cn.ibizlab.pms.core.ibiz.domain;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.math.BigInteger;
import java.util.HashMap;
import java.math.BigDecimal;
import java.text.SimpleDateFormat;
import com.alibaba.fastjson.annotation.JSONField;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.util.ObjectUtils;
import org.springframework.util.DigestUtils;
import cn.ibizlab.pms.util.domain.EntityBase;
import cn.ibizlab.pms.util.annotation.DEField;
import cn.ibizlab.pms.util.enums.DEPredefinedFieldType;
import cn.ibizlab.pms.util.enums.DEFieldDefaultValueType;
import cn.ibizlab.pms.util.helper.DataObject;
import cn.ibizlab.pms.util.enums.DupCheck;
import java.io.Serializable;
import lombok.*;
import org.springframework.data.annotation.Transient;
import cn.ibizlab.pms.util.annotation.Audit;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.baomidou.mybatisplus.annotation.*;
import cn.ibizlab.pms.util.domain.EntityMP;
import com.baomidou.mybatisplus.core.toolkit.IdWorker;

/**
 * 实体[产品汇总表]
 */
@Getter
@Setter
@NoArgsConstructor
@JsonIgnoreProperties(value = "handler")
@TableName(value = "zt_product", resultMap = "ProductSumResultMap")
@ApiModel("产品汇总表")
public class ProductSum extends EntityMP implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 设计缺陷
     */
    @TableField(exist = false)
    @JSONField(name = "designdefect")
    @JsonProperty("designdefect")
    @ApiModelProperty("设计缺陷")
    private Integer designdefect;
    /**
     * 已变更
     */
    @TableField(exist = false)
    @JSONField(name = "changedstorycnt")
    @JsonProperty("changedstorycnt")
    @ApiModelProperty("已变更")
    private Integer changedstorycnt;
    /**
     * 研发中阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "developingstagestoryhours")
    @JsonProperty("developingstagestoryhours")
    @ApiModelProperty("研发中阶段需求工时")
    private Integer developingstagestoryhours;
    /**
     * 总计
     */
    @TableField(exist = false)
    @JSONField(name = "storycnt")
    @JsonProperty("storycnt")
    @ApiModelProperty("总计")
    private Integer storycnt;
    /**
     * 其他
     */
    @TableField(exist = false)
    @JSONField(name = "others")
    @JsonProperty("others")
    @ApiModelProperty("其他")
    private Integer others;
    /**
     * 配置相关
     */
    @TableField(exist = false)
    @JSONField(name = "config")
    @JsonProperty("config")
    @ApiModelProperty("配置相关")
    private Integer config;
    /**
     * 代码错误
     */
    @TableField(exist = false)
    @JSONField(name = "codeerror")
    @JsonProperty("codeerror")
    @ApiModelProperty("代码错误")
    private Integer codeerror;
    /**
     * bug总计
     */
    @TableField(exist = false)
    @JSONField(name = "bugsum")
    @JsonProperty("bugsum")
    @ApiModelProperty("bug总计")
    private Integer bugsum;
    /**
     * 性能问题
     */
    @TableField(exist = false)
    @JSONField(name = "performance")
    @JsonProperty("performance")
    @ApiModelProperty("性能问题")
    private Integer performance;
    /**
     * 结束日期
     */
    @TableField(exist = false)
    @JSONField(name = "end")
    @JsonProperty("end")
    @ApiModelProperty("结束日期")
    private String end;
    /**
     * 已关闭
     */
    @TableField(exist = false)
    @JSONField(name = "closedstorycnt")
    @JsonProperty("closedstorycnt")
    @ApiModelProperty("已关闭")
    private Integer closedstorycnt;
    /**
     * 已关闭阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "closedstagestoryhours")
    @JsonProperty("closedstagestoryhours")
    @ApiModelProperty("已关闭阶段需求工时")
    private Integer closedstagestoryhours;
    /**
     * 测试完毕阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "testedstagestoryhours")
    @JsonProperty("testedstagestoryhours")
    @ApiModelProperty("测试完毕阶段需求工时")
    private Integer testedstagestoryhours;
    /**
     * 未开始阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "waitstagestoryhours")
    @JsonProperty("waitstagestoryhours")
    @ApiModelProperty("未开始阶段需求工时")
    private Integer waitstagestoryhours;
    /**
     * 产品负责人
     */
    @TableField(value = "`po`")
    @JSONField(name = "po")
    @JsonProperty("po")
    @ApiModelProperty("产品负责人")
    private String po;
    /**
     * 测试中阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "testingstagestoryhours")
    @JsonProperty("testingstagestoryhours")
    @ApiModelProperty("测试中阶段需求工时")
    private Integer testingstagestoryhours;
    /**
     * 已立项阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "projectedstagestoryhours")
    @JsonProperty("projectedstagestoryhours")
    @ApiModelProperty("已立项阶段需求工时")
    private Integer projectedstagestoryhours;
    /**
     * 已立项阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "projectedstagestorycnt")
    @JsonProperty("projectedstagestorycnt")
    @ApiModelProperty("已立项阶段需求数量")
    private Integer projectedstagestorycnt;
    /**
     * 已验收阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "verifiedstagestorycnt")
    @JsonProperty("verifiedstagestorycnt")
    @ApiModelProperty("已验收阶段需求数量")
    private Integer verifiedstagestorycnt;
    /**
     * 主键标识
     */
    @DEField(isKeyField = true)
    @TableId(value = "id", type = IdType.AUTO)
    @JSONField(name = "id")
    @JsonProperty("id")
    @ApiModelProperty("主键标识")
    private Long id;
    /**
     * 总工时
     */
    @TableField(exist = false)
    @JSONField(name = "totalhours")
    @JsonProperty("totalhours")
    @ApiModelProperty("总工时")
    private Integer totalhours;
    /**
     * 已计划阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "planedstagestorycnt")
    @JsonProperty("planedstagestorycnt")
    @ApiModelProperty("已计划阶段需求数量")
    private Integer planedstagestorycnt;
    /**
     * 研发中阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "developingstagestorycnt")
    @JsonProperty("developingstagestorycnt")
    @ApiModelProperty("研发中阶段需求数量")
    private Integer developingstagestorycnt;
    /**
     * 激活
     */
    @TableField(exist = false)
    @JSONField(name = "activestorycnt")
    @JsonProperty("activestorycnt")
    @ApiModelProperty("激活")
    private Integer activestorycnt;
    /**
     * Bug数
     */
    @TableField(exist = false)
    @JSONField(name = "bugcnt")
    @JsonProperty("bugcnt")
    @ApiModelProperty("Bug数")
    private Integer bugcnt;
    /**
     * 未开始阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "waitstagestorycnt")
    @JsonProperty("waitstagestorycnt")
    @ApiModelProperty("未开始阶段需求数量")
    private Integer waitstagestorycnt;
    /**
     * 已验收阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "verifiedstagestoryhours")
    @JsonProperty("verifiedstagestoryhours")
    @ApiModelProperty("已验收阶段需求工时")
    private Integer verifiedstagestoryhours;
    /**
     * 草稿
     */
    @TableField(exist = false)
    @JSONField(name = "waitstorycnt")
    @JsonProperty("waitstorycnt")
    @ApiModelProperty("草稿")
    private Integer waitstorycnt;
    /**
     * 测试中阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "testingstagestorycnt")
    @JsonProperty("testingstagestorycnt")
    @ApiModelProperty("测试中阶段需求数量")
    private Integer testingstagestorycnt;
    /**
     * 开始日期
     */
    @TableField(exist = false)
    @JSONField(name = "begin")
    @JsonProperty("begin")
    @ApiModelProperty("开始日期")
    private String begin;
    /**
     * 计划
     */
    @TableField(exist = false)
    @JSONField(name = "plan")
    @JsonProperty("plan")
    @ApiModelProperty("计划")
    private Long plan;
    /**
     * 已发布阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "releasedstagestoryhours")
    @JsonProperty("releasedstagestoryhours")
    @ApiModelProperty("已发布阶段需求工时")
    private Integer releasedstagestoryhours;
    /**
     * 研发完毕阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "developedstagestoryhours")
    @JsonProperty("developedstagestoryhours")
    @ApiModelProperty("研发完毕阶段需求工时")
    private Integer developedstagestoryhours;
    /**
     * 产品名称
     */
    @TableField(value = "`name`")
    @JSONField(name = "name")
    @JsonProperty("name")
    @ApiModelProperty("产品名称")
    private String name;
    /**
     * 研发完毕阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "developedstagestorycnt")
    @JsonProperty("developedstagestorycnt")
    @ApiModelProperty("研发完毕阶段需求数量")
    private Integer developedstagestorycnt;
    /**
     * 测试脚本
     */
    @TableField(exist = false)
    @JSONField(name = "automation")
    @JsonProperty("automation")
    @ApiModelProperty("测试脚本")
    private Integer automation;
    /**
     * 已计划阶段需求工时
     */
    @TableField(exist = false)
    @JSONField(name = "planedstagestoryhours")
    @JsonProperty("planedstagestoryhours")
    @ApiModelProperty("已计划阶段需求工时")
    private Integer planedstagestoryhours;
    /**
     * 安装部署
     */
    @TableField(exist = false)
    @JSONField(name = "install")
    @JsonProperty("install")
    @ApiModelProperty("安装部署")
    private Integer install;
    /**
     * 安全相关
     */
    @TableField(exist = false)
    @JSONField(name = "security")
    @JsonProperty("security")
    @ApiModelProperty("安全相关")
    private Integer security;
    /**
     * 已发布阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "releasedstagestorycnt")
    @JsonProperty("releasedstagestorycnt")
    @ApiModelProperty("已发布阶段需求数量")
    private Integer releasedstagestorycnt;
    /**
     * 测试完毕阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "testedstagestorycnt")
    @JsonProperty("testedstagestorycnt")
    @ApiModelProperty("测试完毕阶段需求数量")
    private Integer testedstagestorycnt;
    /**
     * 标准规范
     */
    @TableField(exist = false)
    @JSONField(name = "standard")
    @JsonProperty("standard")
    @ApiModelProperty("标准规范")
    private Integer standard;
    /**
     * 已关闭阶段需求数量
     */
    @TableField(exist = false)
    @JSONField(name = "closedstagestorycnt")
    @JsonProperty("closedstagestorycnt")
    @ApiModelProperty("已关闭阶段需求数量")
    private Integer closedstagestorycnt;



    /**
     * 设置 [产品负责人]
     */
    public void setPo(String po) {
        this.po = po;
        this.modify("po", po);
    }

    /**
     * 设置 [产品名称]
     */
    public void setName(String name) {
        this.name = name;
        this.modify("name", name);
    }


    @Override
    public Serializable getDefaultKey(boolean gen) {
        return IdWorker.getId();
    }
    /**
     * 复制当前对象数据到目标对象(粘贴重置)
     * @param targetEntity 目标数据对象
     * @param bIncEmpty  是否包括空值
     * @param <T>
     * @return
     */
    @Override
    public <T> T copyTo(T targetEntity, boolean bIncEmpty) {
        this.reset("id");
        return super.copyTo(targetEntity, bIncEmpty);
    }
}


